### PR TITLE
Enable rails/delegate on `.rubocup.yml`

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
enable  rails/delegate on `.rubocup.yml` because of `# TODO` and change some methods to  `delegator`

### CODE REVIEW

**Comment by Bo98** [comment]
I suggest not using Rail's delegate method and using Forwardable's ones instead (you might need to use `def_delegator` instead of `delegate` to satisfy Sorbet in some cases). The less unnecessary dependency on activesupport the better.

File: Library/Homebrew/version/null.rb [hunk 1]
```diff
@@ -77,12 +77,12 @@ def major_minor_patch
       self
     end
 
-    sig { override.returns(Float) }
+    # sig { override.returns(Float) }
```
Comment (Bo98) [inline Library/Homebrew/version/null.rb:80]: This is not ideal. If we're losing typing then I wonder whether it's even worth delegating these.

File: Library/Homebrew/tab.rb [hunk 2]
```diff
@@ -240,9 +240,11 @@ def without?(val)
     !with?(val)
   end
 
+  # rubocop:disable Rails/Delegate
```
Comment (MikeMcQuaid) [inline Library/Homebrew/tab.rb:243]: Out of interest: why disable here?

File: Library/Homebrew/cask/download.rb [hunk 3]
```diff
@@ -45,17 +47,13 @@ def time_file_size(timeout: nil)
...
+    def_delegator :downloader, :clear_cache, :clear_cache
 
     def cached_download
       downloader.cached_location
     end
 
-    def basename
-      downloader.basename
-    end
+    def_delegator :downloader, :basename, :basename
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cask/download.rb:56]:
```suggestion
```

File: Library/Homebrew/cask/download.rb [hunk 4]
```diff
@@ -45,17 +47,13 @@ def time_file_size(timeout: nil)
       downloader.resolved_time_file_size(timeout: timeout)
     end
 
-    def clear_cache
-      downloader.clear_cache
-    end
+    def_delegator :downloader, :clear_cache, :clear_cache
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cask/download.rb:50]:
```suggestion
    def_delegators :downloader, :clear_cache, :basename
```

Although may want to move elsewhere.

File: Library/Homebrew/software_spec.rb [hunk 5]
```diff
@@ -190,9 +190,7 @@ def uses_from_macos_names
     uses_from_macos_elements.flat_map { |e| e.is_a?(Hash) ? e.keys : e }
   end
 
-  def deps
-    dependency_collector.deps
-  end
+  def_delegator :dependency_collector, :deps, :deps
```
Comment (MikeMcQuaid) [inline Library/Homebrew/software_spec.rb:193]: `def_delegators` here too

File: Library/Homebrew/version.rb [hunk 6]
```diff
@@ -70,25 +74,13 @@ def inspect
       "#<#{self.class.name} #{value.inspect}>"
     end
 
-    sig { returns(Integer) }
-    def hash
-      value.hash
-    end
+    def_delegator :value, :hash, :hash
```
Comment (MikeMcQuaid) [inline Library/Homebrew/version.rb:77]: `def_delegators` here too

**Review by MikeMcQuaid — COMMENTED** [review]
Great work so far! I also wonder if it's worth checking existing uses of `delegate`. In `formula.rb` it looks like a bunch of them could better use `def_delegators`, for example?

File: Library/Homebrew/tab.rb (hunk 2 above)
Comment (hyuraku) [reply Library/Homebrew/tab.rb:243]: I commented it out temporarily.
I'll remove these comments and change `delegator`.

**Review by RandomDSdevel — COMMENTED** [review]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;You're a bit inconsistent in a couple of your '`def_delegator`' calls as to whether you explicitly pass a name alias to the function as its third, last argument for it to use or just let it fall back to being taken care of by that parameter's default argument and have the delegator's symbol name match the one for the entity which you're delegating to.  You may well want to pick a single, consistent style there.

**Pull request closed without merging**